ibis/file/csv/client.py

```diff
@@ -65,4 +66,26 @@ def list_databases(self, path=None):
 
 @data_preload.register(CSVTable, (pathlib.Path, str))
 def csv_data_preload_uri_client(table, path, scope=None, **kwargs):
-    return pd.read_csv(str(path), header=0)
+    if scope is not None:
+        usecols = scope.get('usecols')
+    else:
+        usecols = None
+    df = pd.read_csv(str(path), usecols=usecols, header=0)
+    return df
+
+
+@pre_execute.register(ops.Selection)
+def csv_pre_execute(op, **kwargs):
+
+    if not (isinstance(op.table.op(), CSVTable) and op.selections):
+        return {}
+
+    # get our source tables
+    pt = physical_tables(op.table.op())
+    valid = []
+    for t in pt:
+        valid.extend(t.schema.names)
```

Comment:
@cpcloud prob not very idiomatic here....

Reply:
`scope` is really only for mapping expressions to data, not a generic way to passing keyword arguments. we should probably just use `**kwargs` for that kind of stuff.

Reply:
so should I actually do the physical read here then? (and just return ``{op:df}`` I think I tried that, let me try again.